src/Components/MovieList/index.tsx

```diff
@@ -1,18 +1,83 @@
-import React from 'react'
+import React, { useReducer } from 'react'
 import Movie from 'Components/Movie'
-import { MoviesResponseModel } from './models'
+import { MovieListLocalState, MovieListReducerAction, MoviesResponseModel } from './models'
 import MoviesMockData from './movies.json'
-import './styles.scss'
 import { MoviesHeader } from './Header'
+import { Modal } from 'Components'
+import MovieForm from 'Components/MovieForm'
+import { MovieFormAction } from 'Components/MovieForm/models'
+import './styles.scss'
+import { MovieModel } from 'Components/Movie/models'
+
+
+
+const reducer = (state: MovieListLocalState, action: MovieListReducerAction) => {
+    return action.payload
+}
+
+const initialState: MovieListLocalState = {
+    isModalOpen: false,
+    selectedMovie: undefined,
+    currentAction: undefined,
+}
 
 const MovieList: React.FC = () => {
+    const [state, dispatch] = useReducer(reducer, initialState)
+
+    const handleEdit = (movie: MovieModel) => {
+        dispatch({
+            payload: {
+                selectedMovie: movie,
+                currentAction: MovieFormAction.EDIT,
+                isModalOpen: true,
+            },
+        })
+    }
+
+    const handleDelete = (movie: MovieModel) => {
+        dispatch({
+            payload: {
+                selectedMovie: movie,
+                currentAction: MovieFormAction.DELETE,
+                isModalOpen: true,
+            },
+        })
+    }
```

Comment:
it can be simplified to one function:
```js
    const handleChange = (movie: MovieModel, type: MovieFormAction) => {
        dispatch({
            payload: {
                selectedMovie: movie,
                currentAction: type,
                isModalOpen: true,
            },
        })
    }
```

Reply:
My bad...
It was too late, my head didn't work properly :)